src/components/FileUploaderDetails/FileUploaderDetails.types.part.ts

```diff
@@ -73,3 +73,9 @@ export interface FileUploaderDetailsEvent<T> {
    */
   files: Array<T>;
 }
+export interface Translate {
+  /**
+   * Optionally provide custom translate function to be used for unhandled custom error messages
+   */
+  translate?(str: string): string;
```

Comment:
Since this is only used for the error I would call it error?(e: Error): string and place it on the general translations object (it should be consistent with the overall approach of Precise UI to localization / messages).

Reply:
the idea is to introduce a unified way to provide translations to the components which need translations without introducing  new props for each translation (how its currently done in `Accordion`: there are `openLabel` and `closeLabel`: the idea is to have `translate` only which will be able to translate these labels ). not for **errors** rather for any potential text

Reply:
I think `str` should be `any`

Reply:
The unified way we have in form of labels - which is imho much better as it transports right away to a user what is possible and what is not there.

As it is proposed in this PR I would not know what to translate and what not 😕. It would all be a guessing game until the next component comes and shows me that it may also have something to translate 🔮 .

Right now (with labels) it is very explicit and allows from a central point to set the translation globally (instead of forcing the user to do it per component).

❗️ **Just to be clear** that we have labels at all is already a bit of a mistake 🤒 ; normally there should not be anything that shows a default string. But since Precise UI also comes with some quite bulky (or *useful*, depending on the POV) components such labels have been necessary 🚀 .

Reply:
the rule of thumb can be used: you have some text in component, ie "Open details", `translate` will be called with the same text, than:
```
const translate = (str) => {
   if(str === 'Open details') {
        return `Открыть детали`
   }

   return str;
}
<Accordion translate={translate} />
```

Also `Accordion` can export:
```
export type AccordionTranslations = 'Open details' | 'Close details'
```
```
export type Translations = AccordionTranslations | LoaderTranslations | any;
export interface Tranlsate {
  (e: Translations): string
}
```

Reply:
Well, for once I *would* call such a function `getText` or `getLabel` (because `translate` assumes we want to translate, which is only an edge case - usually we only want to override the original wording with our wording).

Second I would not transport "original text", but rather keys (this avoids the duplication problem, which can happen across components and inside components, plus limits the verbosity of this concept).

Finally, I would then not use `string` (or `any`) as the type, but rather either something like `LabelKeys = 'label1' | 'label2' | ...`, or even better some overloads to allow a case with two or more arguments like the described error case, where we have (e.g.,) a key "error" and a custom argument of type `Error`.

Personally, I am still not convinced that this solves a real problem. I think the current approach is much more React-like (we have a prop and the name of the prop is very explicit) plus user-friendly (we can locally override it, or even make the global settings different to already start with a default).

I think there are much severe problems that would need attention than this (still unsure why an inconsistent approach is introduced - either change it completely or go with the existing one).

Reply:
> Finally, I would then not use `string` (or `any`) as the type, but rather either something like `LabelKeys = 'label1' | 'label2' | ...`, or even better some overloads to allow a case with two or more arguments like the described error case, where we have (e.g.,) a key "error" and a custom argument of type `Error`.

I didn't get this one. Could you please provide an example especially for the case when component needs to translate something which is not defined in the component (error code).

Consistency can be fixed. Likely we don't have much cases.

Reply:
Component user wants not to `getText`, `getLabels` but rather set it.

Reply:
> Component user wants not to `getText`, `getLabels` but rather set it.

This is not how APIs are designed imho. It is never about *what the user wants*, because the API does not (and cannot!) know that. The API can only communicated *what it offers* / wants, e.g., the *onClick* is the event *from* the component, but the user *wants* to call an API as a consequence. Still we do not name the event *onCallTheApi*. 

In this case the API offers the user to get the text to display (not saying these are good names, but the get / set discussion is not very useful imho).

(also setting implies that we actually place the value somewhere, when in fact we should only retrieve a value)

Reply:
For me as consumer `precise-ui` is a black box. I want  to set text of `Card` component to show "Открыть" instead of open. What is more logical prop name for me `setText` or `getText`?

Reply:
For me personally `getText`, because I should not set text, but provide text. Either way what does `translate` do? Should I now translate? I have a feeling we go in circles.

This is not - yet - about any naming. It was just an example. I think we should not have this discussion at all, because the whole PR is not sound.

The problem you are facing can be solved with the current approach easily. So what is the problem?

(this already seems to be about taste instead of logic and the technical basis - I would appreciate if we could tackle the technical foundation first before we make this a discussion on names)